manuals/Manual_convert_berthplan_to_events.md

```diff
@@ -0,0 +1,347 @@
+Berth: The berth the vessel will visit<br />
+Timestamp: The TOS keeps 4 timestamps for each visit: Arrival, Departure, Cargo Start and Cargo Completion. Additionally each timestamp is marked as an estimate until it is confirmed by an operator to have happened.
+
+## Sending events
+
+In order to share the berth plan with other parties, the TOS should convert the berth plan into events. <br />
+If we want to send the ETD for visit `95aba704`, the event should be as follows:
+
+##### 95aba704 PTA Berth
+```javascript
+{
+    // A random or hash-based UUID generated by the terminal
+    "uuid": "c556f290-5d17-4d40-bf6d-7471295825d3",
+    // The event format version used
+    "version": "3.1.1",
+    // A string identifying the Gilo terminal as agreed with other participants
+    "source": "XXGIL Terminal",
+    // The event type of this timestamp, see the table below
+    "eventType": "berth.pta.terminal",
+    // The timestamp at which this event was created, in UTC
+    "recordTime": "2018-06-08T7:02:00Z",
+    // The ETD
+    "eventTime": "2018-06-08T12:00:00Z",
+    // The ship identifiers and (optional) name
+    "ship": {
+        "imo": "9500936",
+        "name": "BUKHA"
+    },
+    // Since the terminal is located in the port Gilo with UNLOCODE XXGIL, this will always be XXGIL
+    "port": "XXGIL",
+    // Since our terminal has a link with the local Port Authority system, it knows the portcall identifier and can add this
+    "portcallId": "XXGIL18000144",
+    "location": {
+      // Since our terminal only has berths and always knows the berth, it can always use berth
+      "type": "berth",
+      // Our terminal already has GLNs assigned to all berths, so can use these
+      "gln": "0061414000031",
+      "name": "Gilo 1"
+    },
+    "context": {
+      // Our TOS knows an estimate of the static draught of the ship when it will leave, so can add the optional draught to the event
+      "draught": 1900,
+      // Our TOS has access to mooring information, so can add the optional mooring information to the event
+      "bollardFore": 98,
+      "bollardAft": 150,
+      "orientation": "Port",
+      // The identifier of the berth visit for our terminal, see the berth visit id paragraph below
+      "berthVisitId": "BID-XXGILTerminal-95aba704"
+    }
+}
+```
+
+Note that the event times should be in UTC (Zulu) time.
+Since our terminal is currently in the +02:00 time zone, two hours need to be subtracted from the local time.
+We recommend using a date-time library for this.
+
+### Event types
+For terminals, the following event types will generally be the most interesting. A full list can be found in the specification.
+
+| Event name | ID                           | Explanation
+| ---------- | ---------------------------- | ---------------------------
+| PTA Berth  | berth.pta.terminal           | The planned time the ship will arrive at the berth, before it is moored
```

Comment:
Do we want to add the ETD event types here as well?

Reply:
Oh oops